### PR TITLE
[Spark-483] Add Kerberos principal and secret-based key tab to the CLI, also update libmesos in Dockerfile

### PR DESCRIPTION
**Pull request by ArtRand**
Add `--kerberos-principal` and `--keytab-secret-path` to the CLI, the former must match the key table in the secret indicated by the latter. Tested manually against a kerberized HDFS cluster. 

Todo:
- [x] add flags for secrets and kerberos principal
- [x] test against 1.10 file-based secrets
- [x] test against 1.10 environment based secrets
~test against 1.9 file-based secrets~

### CODE REVIEW

File: dispatcher/cli/dcos_spark/cli.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
     dcos spark run --help
     dcos spark run --submit-args=<spark-args>
                    [--dcos-space=<dcos_space>]
+                   [--kerberos-principal=kerberos_principal]
+                   [--keytab-secret-path=<keytab_secret>]
```
Comment (ArtRand) [inline dispatcher/cli/dcos_spark/cli.py]: Fixed. I also removed the `--tgt` section of the docs, because that doesn't actually exist AFAICT?

**Comment by ArtRand** [comment]
retest this please (strange docker build flake)

File: docker/runit/init.sh [hunk 2]
```diff
@@ -65,6 +65,7 @@ if [[ -f hdfs-site.xml && -f core-site.xml ]]; then
 fi
 
 # Move kerberos config file, as specified by security.kerberos.krb5conf, into place.
+# this only effects the krb5.conf file for the dispatcher
```
Comment (elezar) [inline docker/runit/init.sh]: typo: affects -> effects

File: dispatcher/cli/dcos_spark/cli.py (hunk 1 above)
Comment (susanxhuynh) [inline dispatcher/cli/dcos_spark/cli.py]: @ArtRand There is some mention of `--tgt` in this custom PR: https://github.com/mesosphere/spark/commit/40b05d3f2321c366a070f5894e95d74b074c89c4

Comment (ArtRand) [inline dispatcher/cli/dcos_spark/cli.py]: @susanxhuynh I see, `--tgt` is a custom commit arg that is passed as a `spark-submit` arg. I updated `hdfs.md` with the new protocol.

**Comment by ArtRand** [comment]
Leaving this branch around for reference for a short time

**Pull request merged**